src/Core/tests/UnitTests/PropertyMapperTests.cs

```diff
@@ -9,6 +12,72 @@ namespace Microsoft.Maui.UnitTests
 	[Category(TestCategory.Core, TestCategory.PropertyMapping)]
 	public class PropertyMapperTests
 	{
+		[Fact]
+		public void MapperRetainsInsertionOrder()
+		{
+			// This test ensures the internal Dictionary implementation is not changed.
+
+			// Thanks to the way Dictionary is internally implemented,
+			// when looping through the dictionary entries, and considering _mapper is an append-only dictionary,
+			// the order is guaranteed to be the same as the order mappers were added.
+
+			var mapper = new PropertyMapper<IElement>();
+			var letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+			var insertedProperties = new List<string>();
+			var updatedProperties = new List<string>();
+			for (int i = 0; i < 1000; i++)
+			{
+				var randomPropertyName = string.Join(string.Empty, Enumerable.Range(0, Random.Shared.Next(8, 60))
+					.Select(chars => string.Join(string.Empty,
+						Enumerable.Range(0, chars).Select(_ => letters[Random.Shared.Next(0, letters.Length)]))));
+
+				insertedProperties.Add(randomPropertyName);
+				mapper.Add(randomPropertyName, (h, v) => updatedProperties.Add(randomPropertyName));
+			}
+
+			var keys = mapper.GetKeys().ToList();
+			mapper.UpdateProperties(null!, new Button());
+			Assert.Equal(insertedProperties, keys);
+			Assert.Equal(insertedProperties, updatedProperties);
+		}
+
+		[Fact]
+		public void MapperExecutesChainedKeysFirst()
+		{
+			int counter = 0;
+
+			int background3 = 0;
+			int scale1 = 0;
+			int scale3 = 0;
+			int zindex2 = 0;
+			int zindex3 = 0;
+
+			var mapper1 = new PropertyMapper<IView>
+			{
+				[nameof(IView.Scale)] = (r, v) => scale1 = ++counter
+			};
+
+			var mapper2 = new PropertyMapper<IView>
+			{
+				[nameof(IView.ZIndex)] = (r, v) => zindex2 = ++counter
+			};
+
+			var mapper3 = new PropertyMapper<IButton>(mapper2, mapper1)
+			{
+				[nameof(IView.Background)] = (r, v) => background3 = ++counter,
+				[nameof(IView.Scale)] = (r, v) => scale3 = ++counter,
+				[nameof(IView.ZIndex)] = (r, v) => zindex3 = ++counter,
+			};
+
+			mapper3.UpdateProperties(null!, new Button());
+
+			Assert.Equal(0, scale1);
+			Assert.Equal(0, zindex2);
```

Comment:
Why are both 0 when they are all using the same incrementing counter?

Reply:
`0` means those mappers didn't run, which is one of the things we want to verify

Reply:
Ah, makes so much sense now that I am rested. lol, rest seems to have value in the human existence!